### PR TITLE
compiler: fix stackPointerCeil calculation

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: internal/engine/compiler/compiler_value_location.go [hunk 1]
```diff
@@ -201,10 +201,10 @@ func (v *runtimeValueLocationStack) push(loc *runtimeValueLocation) {
 	} else {
 		v.stack[v.sp] = loc
 	}
+	v.sp++
```
Comment (mathetake) [inline internal/engine/compiler/compiler_value_location.go:204]: so stackPointerCeil must be set after sp is incremented :D

Comment (mathetake) [reply internal/engine/compiler/compiler_value_location.go:204]: as we skip the stack grow if `len(valueStack) >=  basePointer+stackPointerCeil`.

Comment (codefromthecrypt) [reply internal/engine/compiler/compiler_value_location.go:204]: sounds like a comment for the code :D

Comment (mathetake) [reply internal/engine/compiler/compiler_value_location.go:204]: done

**Review by codefromthecrypt — APPROVED** [review]
nice

**Pull request merged**